eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java

```diff
@@ -308,9 +314,12 @@ public InstanceInfo(InstanceInfo ii) {
         UNKNOWN;
 
         public static InstanceStatus toEnum(String s) {
-            for (InstanceStatus e : InstanceStatus.values()) {
-                if (e.name().equalsIgnoreCase(s)) {
-                    return e;
+            if (s != null) {
+                try {
+                    return InstanceStatus.valueOf(s.toUpperCase());
+                } catch (IllegalArgumentException e) {
+                    // ignore and fall through to unknown
+                    if (logger.isDebugEnabled()) logger.debug("illegal argument supplied to InstanceStatus.valueOf: {}, defaulting to {}", s, UNKNOWN);
```

Comment:
In this particular case couldn't you just remove the condition? It is already using placeholders to avoid substituting in the string if debug is not enabled and the logging library will be doing the same check internally.